openml/_api_calls.py

```diff
@@ -103,21 +104,32 @@ def _download_text_file(
         except FileNotFoundError:
             pass
 
+    n_retries = cast(int, config.connection_n_retries)
+    wait_time = 0.2
+    raise_error = None
     logging.info("Starting [%s] request for the URL %s", "get", source)
     start = time.time()
-    response = __read_url(source, request_method="get")
-    downloaded_file = response.text
-
-    if md5_checksum is not None:
-        md5 = hashlib.md5()
-        md5.update(downloaded_file.encode("utf-8"))
-        md5_checksum_download = md5.hexdigest()
-        if md5_checksum != md5_checksum_download:
-            raise OpenMLHashException(
-                "Checksum {} of downloaded file is unequal to the expected checksum {}.".format(
-                    md5_checksum_download, md5_checksum
-                )
-            )
+    for retry in range(n_retries):
+        response = __read_url(source, request_method="get")
```

Comment:
I believe `__read_url` internally already does retries too, so we allow up to `n_retries**2` retries? Should we consider propagating the checksum down a few levels?